api/tests/core/users/test_api.py

```diff
@@ -1769,3 +1775,166 @@ def test_unsuspended_account(self) -> None:
 
         with assert_num_queries(1):
             assert not list(users_api.get_suspended_upon_user_request_accounts_since(5))
+
+
+class AnonymiseNonProNonBeneficiaryUsersTest:
+    def import_iris(self):
+        path = DATA_DIR / "iris_min.7z"
+        geography_api.import_iris_from_7z(str(path))
+
+    def test_anonymise_non_pro_non_beneficiary_users(self) -> None:
+        user_to_anonymize = users_factories.UserFactory(
+            firstName="user_to_anonymize",
+            lastConnectionDate=datetime.datetime.utcnow() - datetime.timedelta(days=((3 * 365) + 1)),
+            validatedBirthDate=datetime.datetime.today(),
+        )
+        user_too_new = users_factories.UserFactory(
+            firstName="user_too_new",
+            lastConnectionDate=datetime.datetime.utcnow() - datetime.timedelta(days=((3 * 365) - 1)),
+        )
+        user_never_connected = users_factories.UserFactory(firstName="user_never_connected", lastConnectionDate=None)
+        user_beneficiary = users_factories.BeneficiaryGrant18Factory(
+            firstName="user_beneficiary",
+        )
+        user_underage_beneficiary = users_factories.UnderageBeneficiaryFactory(
+            firstName="user_underage_beneficiary",
+        )
+        user_pro = users_factories.ProFactory(
+            firstName="user_pro",
+        )
+        user_pass_culture = users_factories.ProFactory(
+            firstName="user_pass_culture", email="user_pass_culture@passculture.app"
+        )
+        user_anonymized = users_factories.ProFactory(
+            firstName="user_anonymized", email="user_anonymized@anonymized.passculture"
+        )
+
+        self.import_iris()
+        iris = geography_models.IrisFrance.query.first()
+
+        with mock.patch("pcapi.core.users.api.get_iris_from_address", return_value=iris):
+            users_api.anonymise_non_pro_non_beneficiary_users(force=False)
+
+        # those profile should not have been anonymized
```

Comment:
```suggestion
        # these profiles should not have been anonymized
```
`those`, c'est quand ils sont loins, et là ils sont pas très loins en vrai

Reply:
3 ans c'est loin dans le temps :p (mais tu as raison)